### PR TITLE
Feature/line items

### PR DESCRIPTION
**Pull request by amonliza**
Line items mapping

### CODE REVIEW

File: src/server/acl.json [hunk 1]
```diff
@@ -37,6 +37,15 @@
                     "de": "PDF Mapping - to be replaced with own URL!"
                 }
             },
+            {
+                "type": "rest",
+                "resourceId": "^/api/pdf2invoice/test$",
```
Comment (nczeranka) [inline src/server/acl.json]: Even it is only meant for test (storing a pdf on the blob for tests), please add a meaningful name for the endpoint.

File: src/server/acl.json [hunk 2]
```diff
@@ -37,6 +37,15 @@
                     "de": "PDF Mapping - to be replaced with own URL!"
                 }
             },
+            {
```
Comment (nczeranka) [inline src/server/acl.json]: Creation of test data for pdf2invoice should reside in pdf2invoice.

File: src/client/api/Pdf2invoice.js [hunk 3]
```diff
@@ -6,15 +6,51 @@ class Pdf2invoice
...
                     if (response.body) {
                       var json = response.body.jsonData;
+                      console.log(json);
                       for (var field of json.fields) {
                         jsonData[field.fieldName] = field.fieldValue
                       }
+                      if (json.lineItems) {
+                        jsonData["lineItems"] = mapLineItems(json.lineItems);
+                      }
                       console.log(jsonData);
```
Comment (nczeranka) [inline src/client/api/Pdf2invoice.js]: see above.

File: src/server/acl.json [hunk 4]
```diff
@@ -37,6 +37,15 @@
...
+                "type": "rest",
+                "resourceId": "^/api/pdf2invoice/test$",
+                "actions": ["view"],
+                "description": {
+                    "en": "Store test file.",
+                    "de": ""
```
Comment (nczeranka) [inline src/server/acl.json]: If you don't have a description for the language (here "de"), then omit the complete key-value pair. Else we won't have a fall back to the English description, but an empty German one.

File: src/client/containers/SalesInvoiceEditor.js [hunk 5]
```diff
@@ -403,7 +404,9 @@ class SalesInvoiceEditor extends Components.ContextComponent
         // errors, salesInvoiceItems, etc. are reused and therefore the emptyFormState already consists objects with the data!
         resetedInvoiceState.errors = {};
         resetedInvoiceState.warnings = {};
-        resetedInvoiceState.salesInvoiceItems = [];
+        if (this.state.jsonData['lineItems']) {
```
Comment (nczeranka) [inline src/client/containers/SalesInvoiceEditor.js]: this.state.jsonData could be undefined -> error.

File: src/client/api/Pdf2invoice.js (hunk 3 above)
Comment (amonliza) [reply src/client/api/Pdf2invoice.js]: I changed the method name as it's real aim is to fill empty properties in items (when the result from pdf2invoice doesn't contain them - sending empty values from service would be rather inefficient especially when there is a lot of items).

**Comment by amonliza** [comment]
Requested changes done.
+ Added line items highlight

**Comment by amonliza** [comment]
Tested on develop.

**Pull request merged**